packages/nimble-components/src/wafer-map/modules/rendering.ts

```diff
@@ -11,17 +12,21 @@ export class RenderingModule {
     public constructor(private readonly wafermap: WaferMap) {}
 
     public updateSortedDiesAndDrawWafer(): void {
-        this.dies = this.wafermap.dataManager.diesRenderInfo.sort((a, b) => {
-            if (a.fillStyle > b.fillStyle) {
-                return 1;
-            }
-            if (b.fillStyle > a.fillStyle) {
-                return -1;
-            }
+        if (this.wafermap.dataManager instanceof DataManager) {
+            this.dies = this.wafermap.dataManager.diesRenderInfo.sort(

```

Comment:
Calling sort will [mutate `diesRenderInfo` in place](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort). I think that is surprising that rendering will mutate values in thee dataManager.

Maybe it should be a shallow copy, i.e. `[...this.wafermap.dataManager.diesRenderInfo].sort()`?
Or we should be explicit about mutating diesRenderInfo within dataManager instead of in the renderer.

Reply:
this appears changed because the type check of the data manager. this is the old rendering code which will get replaced